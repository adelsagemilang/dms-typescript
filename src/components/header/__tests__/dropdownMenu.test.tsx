import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import DropdownMenu from '../dropdownMenu';

describe('Dropdown Menu Header Tests', () => {
  it('renders correctyly', () => {
    const wrapper = shallow(<DropdownMenu />);
    expect(wrapper).toMatchSnapshot();
  });

  it('called logout on click', () => {
    const mockCallback = sinon.spy();
    const wrapper = shallow(<DropdownMenu logout={mockCallback} />);
    wrapper.find('.logout').simulate('click');
    expect(mockCallback).toHaveProperty('callCount', 1);
  });
});
