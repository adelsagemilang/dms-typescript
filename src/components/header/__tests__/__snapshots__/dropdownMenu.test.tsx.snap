// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Menu Header Tests renders correctyly 1`] = `
<withHelpersModifiers(NavbarDropdown)>
  <withHelpersModifiers(Columns)
    className="navbar-dropdown-items"
  >
    <withHelpersModifiers(Column)>
      <Link
        to="/"
      >
        <withHelpersModifiers(NavbarItem)
          className="navbar-item-icon"
        >
          <img
            alt="icon-home"
            src="/../images/icon-home.png"
          />
        </withHelpersModifiers(NavbarItem)>
        <div
          className="navbar-item-title"
        >
          Home
        </div>
      </Link>
    </withHelpersModifiers(Column)>
    <withHelpersModifiers(Column)>
      <div
        className="logout"
      >
        <Link
          to="/logout"
        >
          <withHelpersModifiers(NavbarItem)
            className="navbar-item-icon"
          >
            <img
              alt="icon-logout"
              src="/../images/logout.svg"
            />
          </withHelpersModifiers(NavbarItem)>
          <div
            className="navbar-item-title"
          >
            Log Out
          </div>
        </Link>
      </div>
    </withHelpersModifiers(Column)>
  </withHelpersModifiers(Columns)>
</withHelpersModifiers(NavbarDropdown)>
`;
